extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3DataSegmentKiller.java

```diff
@@ -150,7 +151,14 @@ private boolean deleteKeysForBucket(
             s3Bucket,
             keysToDeleteStrings
         );
-        s3Client.deleteObjects(deleteObjectsRequest);
+        RetryUtils.retry(
+            () -> {
+              s3Client.deleteObjects(deleteObjectsRequest);
+              return null;
+            },
+            exception -> (exception instanceof MultiObjectDeleteException) || S3Utils.S3RETRY.apply(exception),
```

Comment:
should the `MultiObjectDeleteException` logic be pushed into `S3RETRY` predicate?

Reply:
done